### ISSUE TITLE
Funding Choices discussion & research

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardFilters#5278]
### How it works
To check it try navigating to androidpolice.com (or any other website https://support.google.com/contributor/answer/7324995) with a US VPN.

You'll see an alert message like this:
![andy rubin s essential raises 300 million investment valuation set at around 1 billion for the startup 2017-06-08 13-24-36](https://user-images.githubusercontent.com/5947035/26924285-dd7c8da4-4c4d-11e7-9c01-b3644a18b61c.png)

You can view a few articles for free and then either allow ads or buy an ad removal pass and pay $0.01-$0.04 per article.

### Below are my thoughts on this topic

"Funding choices" is not just an adblocking-blocker, it is a legit alternative for funding websites without seeing ads, so we should not simply circumvent it.

Instead, we should try to improve users experience.

For example, here's what we can do:

1. Integrate somehow with this contributor window and allow seamless disabling/enabling of AG for the website.
2. Alert user about ads he'll see if he opts to disable AG. How many ads are there, how many third-party requests will be allowed, how will it affect website performance, trackers/etc.
3. Provide a semi-unblocking option, which will continue to block tracking stuff and will clear cookies and block fingerprinting from the unblocked domains.
4. Reformat that alert window so that it was not that annoying
5. If a user is not ready to pay or unblock ads, we might help him find an alternative.

### What do we need now

1. I need your feedback on the Funding Choices. What's your opinion, what're your thoughts, etc.
2. We need to conduct a research on the technical side of it. How does it work, how does it track users, etc.

**Comment by ameshkov** [comment]
@seanl-adg regarding the research, I know that you're good at this kind of things. Could you please do it and post the result here?

**Comment by ameshkov** [comment]
@seanl-adg I feel that it'd be better to discuss it here.

**Why start this discussion now and not when SourcePoint or others emerged?**
Because I suppose that funding choices will become very popular pretty soon, and we'd better be ready for this.

**Why being "soft" on FC?**
Frankly, I'd like us to be "soft" by default on services that provide an alternative for people, who do not want to see ads or share their data with tracking systems. You either see ads or pay, that's fair.

AdGuard, as a filtering software, is very powerful and indeed you can easily circumvent FC. However, I find that circumventing it **by default** is a wrong way.

Here is why:

1. Content creators deserve to be paid.
2. If "choices" model is sabotaged/dismantled, we risk getting into a situation where hard paywall is the only viable option left.

**Comment by TPS** [comment]
Could this be coordinated somehow w/ our choice in-app for the AG Acceptable Ads filter (or whatever it's called now)?

**Comment by ameshkov** [comment]
> AG Acceptable Ads

AG does not have any acceptable ads, this is a very short list for those who want to see search ads in google/bing/yahoo.

**Comment by ameshkov** [comment]
Further discussion in Slack helped me realize that there's a difference in how we perceive FC.

Sean considers it another kind of anti-abortion, so being soft on it may look like an "acceptable anti-adblock" program, and frankly, that's not what I want, had enough of that confusion with the list unblocking search ads and self-promo (which actually aren't blocked on purpose even by easylist).

My understanding is that FC is a new kind of paywall, hence the proposed different attitude towards it.

I realize that the topic is controversial and I haven't yet thought it through, just throwing ideas.

**Comment by TPS** [comment]
>AG does not have any acceptable ads, this is a very short list for those who want to see search ads in google/bing/yahoo

>being soft on it may look like an "acceptable anti-adblock" program, and frankly, that's not what I want
My understanding is that FC is a new kind of paywall, hence the proposed different attitude towards it.

I suggest that there's considerable overlap in the _users_ who may be interested in the 2 categories.  To me, it's 2 similar methods to accomplish the same thing: to monetize in such a way that I (the user) has transparent choice.  In terms of apps, it'd be a dev choosing to do Amazon Unlimited vs. freemium model.  So expanding the current settings to handle this would be uninstrusive to us, but we'd perceive it as opt-in value-added, rather than opt-out Acceptable Ads (which caused all that problem w/ AB+).

**Comment by ameshkov** [comment]
@TPS yeah, that's exactly what I was thinking.

Our purpose is not to fight paywalls (which might also be illegal) but to improve user experience, and I am just considering options.

**Comment by TPS** [comment]
> not to fight paywalls (which might also be illegal)

Soon, if not now, it'll be equally illegal to block ads in same places as bypassing paywalls. 🤷‍♂️

**Comment by sandybeachit** [comment]
There is one potential downside on allowing these types of soft systrms to exist, the view and or pay for site is not a bad idea so to be clear I am not against this idea.

The only possible issue is if it became popular enough scammers might start using it and create fake news websites or just scams to get you to pay or to get your credit card info.

I'm not saying that adguard should circumvent this paywall or just allow ads, I'm just saying eventually we may see scammers start to use this to scam people, so any decision made here should take into account what happens if scammers use this system?

Also what if a scammer hacks a legit website and installs a soft paywall like this? Do you block the domain? Get around the fake paywall to prevent the scam affecting users etc.

I'm just saying this needs to be considered so whatever decision you make here when you come across scammers you know what your answer to them is.

**Comment by sandybeachit** [comment]
Also @TPS regarding making blocking ads illegal ithat might be a per country issue.

Sure as in the UK the mobile telephone company called three did a 24 hour test on their network blocking all adverts (like your adguard DNS) with the idea to eventually block all adverts on their network.

Unless advertiser's clear up their adverts and make adverts 100% not contain malware, phishing then we have a legit reason to block the adverts.

We can't get malware/phishing from adverts on the radio, TV or from my android based radio apps like rock radio.com so I don't try to bypass/skip/block adverts.

At the end of the day if advertiser's listened to their customers who viewed the ads and didn't piss them off adguard, ublock, adblocker plus etc would not exist.

Also what if you have epilepsy? Blocking adverts is fine in that case, it pisses me off when I'm reading something and I have e a flashing advert or advert that changes every second two centimetres from my face.

Sorry for my rant but in short even Google won't try a lawsuit against adblocker in a real court as there is so much that will get dragged up and then the question is who pays if my kid has a seizure due to your advert? Who pays if malware is on my machine from your advert that got hacked etc etc.

I think we're a long way off from a acourt case, maybe once advertiser's play nice lol

**Comment by ameshkov** [comment]
FC seems to be more dead than alive after all
